### PR TITLE
Use `as_millis()` in `connect_timeout()`

### LINKED ISSUE #68168: Modify the calculation of timeout in TcpStream::connect_timeout
https://github.com/rust-lang/rust/blob/3ebcfa1451cfedc13a07e6353d8ade9742dfdc2a/src/libstd/sys/unix/net.rs#L143-L146

Because `TcpStream::connect_timeout` is implemented in the earlier version, `Duration.as_millis` is not yet stable. However, `Duration.as_millis` is already stable at v1.33, where the same calculation is made.

https://github.com/rust-lang/rust/blob/3ebcfa1451cfedc13a07e6353d8ade9742dfdc2a/src/libcore/time.rs#L338-L343

So this paragraph can directly `let mut timeout = timeout.as_millis() as u64;`.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #68168

### CODE REVIEW

File: src/libstd/sys/unix/net.rs [hunk 1]
```diff
@@ -140,10 +140,7 @@ impl Socket {
             }
 
             let timeout = timeout - elapsed;
-            let mut timeout = timeout
-                .as_secs()
-                .saturating_mul(1_000)
-                .saturating_add(timeout.subsec_nanos() as u64 / 1_000_000);
+            let mut timeout = timeout.as_millis() as u64;
```
Comment (sfackler) [inline src/libstd/sys/unix/net.rs]: This does not maintain the saturating behavior of the original implementation.

Comment (JohnTitor) [reply src/libstd/sys/unix/net.rs]: So, this approach isn't correct? If so, we should just close that issue, I think.

Comment (sfackler) [reply src/libstd/sys/unix/net.rs]: The conversion should saturate to `u64::max_value()` if it would be unrepresentable.

Comment (JohnTitor) [reply src/libstd/sys/unix/net.rs]: Hm, the casting should saturate `as_millis()` value. Should it be saturated in stages like the current behavior?

Comment (sfackler) [reply src/libstd/sys/unix/net.rs]: `as` casts do not saturate, they just discard the high bits.

Comment (JohnTitor) [reply src/libstd/sys/unix/net.rs]: > If so, we should just close that issue, I think.

Then I think we should just close this PR and that issue as above my comment, right?

Comment (JohnTitor) [reply src/libstd/sys/unix/net.rs]: Ah, sorry, I just got what you mean, will fix...

Comment (sfackler) [reply src/libstd/sys/unix/net.rs]: Seems reasonable.

Comment (JohnTitor) [reply src/libstd/sys/unix/net.rs]: Okay, closing.

**Comment by JohnTitor** [comment]
@sfackler As suggested on the issue, `u64::try_from(timeout.as_millis()).unwrap_or(u64::max_value())`seems acceptable, what about?

**Pull request closed without merging**